src/component/app/index.tsx

```diff
@@ -1,20 +1,49 @@
-import { Route, Routes } from "react-router-dom";
+import { Navigate, Route, Routes } from "react-router";
 
+import Profile from "component/page/profile";
 import Signin from "component/page/signin";
 import { AuthContext } from "context/AuthContext";
+import { IUserContext, UserContext } from "context/UserContext";
 import { useAuth } from "module/auth";
+import { useCurrentUser } from "module/hook/user";
 
-import { AuthenticatedRoute } from "./AuthenticatedRoute";
+import { RegisteredUserRoute } from "./RegisteredUserRoute";
 
 export function App() {
-  const auth = useAuth();
+  const authContext = useAuth();
+
+  const { data: userData, status: userStatus } = useCurrentUser({
+    token: authContext.token,
+  });
+
+  if (authContext.loading) {
+    return <span>Checking auth...</span>;
+  }
+
+  const authRequired =
+    !authContext.loading &&
+    !authContext.authenticated &&
+    window.location.pathname !== "/signin"; // Avoid circular redirect when already on signin page
```

Comment:
Here is one of the main fixes--prevent redirecting to /signin after authentication succeeds because it causes infinite redirects.